File: 1_data_manipulation/dplyr/09_top_n.R
# 

# # Getting the rows with the top _n_ values in a column of an R data frame

# Load packages and read inventory data
library(readr)
library(dplyr)
inventory <- read_tsv("data/inventory/data.txt")
inventory

# Example: What's the least expensive game in each shop?

inventory %>%
  group_by(shop) %>%
  top_n(1, price)

# To return the _most_ expensive game in each shop, set
# -1 as the `n` argument to `top_n()`

# It is also possible to achieve this using dplyr's
# [windowed rank functions](https://dplyr.tidyverse.org/reference/ranking.html)

File: 1_data_manipulation/pandas/09_top_n.py
# 

# # Getting the rows with the top _n_ values in a column of a pandas DataFrame

# Import modules and read inventory data
import numpy as np
import pandas as pd
inventory = pd.read_table('data/inventory/data.txt', sep="\t")
inventory

# Example: What's the least expensive game in each shop?

inventory \
  .sort_values('price', ascending=True) \
  .groupby('shop') \
  .head(1)

# To return find the _most_ expensive game in each shop,
# set `ascending=False`.
  
# Optional:
# Add `.sort_values('shop')` to arrange by shop
# Add `.reset_index(drop=True)` to reset the indices
